sql: update doc for split partition table (#2344)
pingcap/docs: reference/sql/statements/split-region.md

`BY value_list…` specifies a series of points manually, based on which the current Region is spilt. It is suitable for scenarios with unevenly distributed data.

The following example shows the result of the `SPLIT` statement:

```sql
+--------------------+----------------------+
| TOTAL_SPLIT_REGION | SCATTER_FINISH_RATIO |
+--------------------+----------------------+
| 4                  | 1.0                  |
+--------------------+----------------------+
```

* `TOTAL_SPLIT_REGION`: the number of newly split Regions.

> * `SCATTER_FINISH_RATIO`: the scattering rate of the newly split Regions. `1.0` means that all Regions are scattered. `0.5` means that only half of the Regions are scattered and the rest are being scattered.

Reviewer: ```suggestion
* `SCATTER_FINISH_RATIO`: the completion rate of scattering for newly split Regions. `1.0` means that all Regions are scattered. `0.5` means that only half of the Regions are scattered and the rest are being scattered.
```

region4  [("c", "")                    , maxIndexValue               )
```

### Split Regions for partitioned tables

Splitting Regions for partitioned tables is the same as splitting Regions for ordinary tables. The only difference is that the same split operation is performed for every partition.

+ The syntax of even split:

    {{< copyable "sql" >}}

    ```sql
    SPLIT [PARTITION] TABLE t [PARTITION] [(partition_name_list...)] [INDEX index_name] BETWEEN (lower_value) AND (upper_value) REGIONS region_num
    ```

+ The syntax of uneven split:

    {{< copyable "sql" >}}

    ```sql
    SPLIT [PARTITION] TABLE table_name [PARTITION (partition_name_list...)] [INDEX index_name] BY (value_list) [, (value_list)] ...
    ```

#### Examples

1. Create a partitioned table `t`.

    {{< copyable "sql" >}}

    ```sql
    create table t (a int,b int,index idx(a)) partition by hash(a) partitions 2;
    ```

>     After creating the table `t`, a Region is split for each partition. Use the `SHOW TABLE REGIONS` syntax to see the Region of this table:

Reviewer: ```suggestion
    After creating the table `t`, a Region is split for each partition. Use the `SHOW TABLE REGIONS` syntax to view the Regions of this table:
```

    | REGION_ID | START_KEY | END_KEY | LEADER_ID | LEADER_STORE_ID | PEERS            | SCATTERING | WRITTEN_BYTES | READ_BYTES | APPROXIMATE_SIZE(MB) | APPROXIMATE_KEYS |
    +-----------+-----------+---------+-----------+-----------------+------------------+------------+---------------+------------+----------------------+------------------+
    | 1978      | t_1400_   | t_1401_ | 1979      | 4               | 1979, 1980, 1981 | 0          | 0             | 0          | 1                    | 0                |
    | 6         | t_1401_   |         | 17        | 4               | 17, 18, 21       | 0          | 223           | 0          | 1                    | 0                |
    +-----------+-----------+---------+-----------+-----------------+------------------+------------+---------------+------------+----------------------+------------------+
    ```

2. Use the `SPLIT` syntax to split a Region for each partition. In the following example, four Regions are split in the range of `[0,10000]`.

    {{< copyable "sql" >}}

    ```sql
    split partition table t between (0) and (10000) regions 4;
    ```

> 3. Use the `SHOW TABLE REGIONS` syntax to see the Region of this table again. You can see that this table now has ten Regions, each partition with five Regions, four of which are the row data and one is the index data.

Reviewer: ```suggestion
3. Use the `SHOW TABLE REGIONS` syntax to view the Regions of this table again. You can see that this table now has ten Regions, each partition with five Regions, four of which are the row data and one is the index data.
```

    | 2002      | t_1401_r_5000 | t_1401_r_7500 | 2003      | 4               | 2003, 2023, 2022 | 0          | 141           | 0          | 1                    | 0                |
    | 6         | t_1401_r_7500 |               | 17        | 4               | 17, 18, 21       | 0          | 601           | 0          | 1                    | 0                |
    | 1986      | t_1401_       | t_1401_r      | 1989      | 5               | 1989, 2018, 2019 | 0          | 123           | 0          | 1                    | 0                |
    +-----------+---------------+---------------+-----------+-----------------+------------------+------------+---------------+------------+----------------------+------------------+
    ```

4. You can also split Regions for the index of each partition. For example, you can split the `[1000,10000]` range of the `idx` index into two Regions:

    {{< copyable "sql" >}}

    ```sql
    split partition table t index idx between (1000) and (10000) regions 2;
    ```

#### Split Region for a single partition

> You can specify the partition to be split. See the following steps for example:

Reviewer: ```suggestion
You can specify the partition to be split. See the following usage example:
```

1. Create a partitioned table:

    {{< copyable "sql" >}}

    ```sql
    create table t ( a int, b int, index idx(b)) partition by range( a ) (
        partition p1 values less than (10000),
        partition p2 values less than (20000),
        partition p3 values less than (MAXVALUE) );
    ```

2. Split two Regions in the `[0,10000]` range of the `p1` partition:

    {{< copyable "sql" >}}

    ```sql
    split partition table t partition (p1) between (0) and (10000) regions 2;
    ```

3. Split two Regions in the `[10000,20000]` range of the `p2` partition:

    {{< copyable "sql" >}}

    ```sql
    split partition table t partition (p2) between (10000) and (20000) regions 2;
    ```

> 4. Use the `SHOW TABLE REGIONS` syntax to see the Region of this table:

Reviewer: From the results you can see there are multiple Regions.
```suggestion
4. Use the `SHOW TABLE REGIONS` syntax to view the Regions of this table:
```

region4  [("c", "")                    , maxIndexValue               )
```

### Split Regions for partitioned tables

> Splitting Regions for partitioned tables is the same as splitting Regions for normal tables. The only difference is that the split operation is performed for every partition.

Reviewer: ```suggestion
Splitting Regions for partitioned tables is the same as splitting Regions for ordinary tables. The only difference is that the same split operation is performed for every partition.
```

+ The syntax of even split:

    {{< copyable "sql" >}}

    ```sql
    SPLIT [PARTITION] TABLE t [PARTITION] [(partition_name_list...)] [INDEX index_name] BETWEEN (lower_value) AND (upper_value) REGIONS region_num
    ```

+ The syntax of uneven split:

    {{< copyable "sql" >}}

    ```sql
    SPLIT [PARTITION] TABLE table_name [PARTITION (partition_name_list...)] [INDEX index_name] BY (value_list) [, (value_list)] ...
    ```

#### Examples

> Create a partitioned table `t`:

Reviewer: More commits are made to https://github.com/pingcap/docs-cn/pull/2810. Please refer to the latest version and update this PR accordingly.
